### PR TITLE
implement fallbacks and preferential reordering for non-strict calls

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,23 +1,21 @@
...
 python:
-  - "2.7"
-  - "3.3"
-  - "3.4"
-  - "3.5"
-  - "nightly" # currently points to 3.6-dev
+- '2.7'
+- '3.3'
+- '3.4'
+- '3.5'
```
Comment (izapolsk) [inline .travis.yml]: 3.7 - 3.8 ?

Comment (RonnyPfannschmidt) [reply .travis.yml]: Good point this year

Comment (izapolsk) [reply .travis.yml]: sorry, I noticed your PR only today.

Comment (RonnyPfannschmidt) [reply .travis.yml]: i#ll take a look when i recovered a bit more from he cold, right now im mush

**Comment by RonnyPfannschmidt** [comment]
discarding this approach

**Pull request closed without merging**